### PR TITLE
Add support to override AWS auth config from cluster create request i…

### PR DESCRIPTION
**Pull request by colin014**
…n case of Amazon EKS

| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Add support to override AWS auth config from cluster create request in case of Amazon EKS

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] OpenAPI and Postman files updated (if needed)

### CODE REVIEW

**Comment by bonifaido** [comment]
Tested with:
```json
banzai cluster create <<EOF
{
  "name": "some-shady-business",
  "location": "eu-west-1",
  "cloud": "amazon",
  "secretName": "aws",
  "properties": {
    "eks": {
      "authConfigMap": "apiVersion: v1\ndata:\n  mapUsers: |\n    - userarn: userarn: arn:aws:iam::{{AccountID}}:user/kracser.nandor\n      username: kracser-nandor\n      groups:\n        - system:masters\n\n  name: aws-auth\n  namespace: kube-system",
      "version": "1.15.10",
      "nodePools": {
        "pool1": {
          "count": 3,
          "minCount": 3,
          "maxCount": 5,
          "autoscaling": true,
          "instanceType": "t2.medium"
        }
      }
    }
  }
}
EOF
```

File: internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go [hunk 1]
```diff
@@ -162,22 +164,31 @@ func (a *BootstrapActivity) Execute(ctx context.Context, input BootstrapActivity
...
 	}
 
-	_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Create(&awsAuthConfigMap)
+	_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Create(mergedConfigMap)
 	if k8serr.ReasonForError(err) == metav1.StatusReasonAlreadyExists {
-		_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Update(&awsAuthConfigMap)
+		_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Update(mergedConfigMap)
 		if err != nil {
-			return nil, errors.WrapIfWithDetails(err, "failed to create config map", "configmap", awsAuthConfigMap.Name)
+			return nil, errors.WrapIfWithDetails(err, "failed to create config map", "configmap", mergedConfigMap.Name)
```
Comment (pregnor) [inline internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go:188]: **Question**: shouldn't this error message be `failed to **update** config map`?

File: internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go [hunk 2]
```diff
@@ -303,3 +314,50 @@ func userNameFromArn(arn string) string {
 
 	return arn[idx+1:]
 }
+
+func mergeAuthConfigMaps(defaultConfigMap, inputConfigMap string) (*v1.ConfigMap, error) {
+	var defaultAWSConfigMap v1.ConfigMap
+	_, _, err := scheme.Codecs.UniversalDeserializer().Decode([]byte(defaultConfigMap), nil, &defaultAWSConfigMap)
+	if err != nil {
+		return nil, errors.WrapIf(err, "failed to decode config map")
```
Comment (pregnor) [inline internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go:322]: **Optional**: a qualifier here and 6 lines below such as `default` and `input` in the error message **MAY** help differentiate between issues with any of the config maps.

File: internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go (hunk 1 above)
Comment (bonifaido) [reply internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go:188]: Thanks, fixing it!

File: internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go (hunk 2 above)
Comment (bonifaido) [reply internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go]: Thanks fixed, good idea.

**Pull request merged**